internal/cli/core/search.go

```diff
@@ -108,15 +111,24 @@ func (sr searchResults) String() string {
 		t := table.New()
 		t.SetHeader(tr("ID"), tr("Version"), tr("Name"))
 		for _, platform := range sr.platforms {
-			name := ""
-			if latest := platform.GetLatestRelease(); latest != nil {
-				name = latest.Name
-			}
-			if platform.Deprecated {
-				name = fmt.Sprintf("[%s] %s", tr("DEPRECATED"), name)
+			// When allVersions is not requested we only show the latest compatible version
+			if !sr.allVersions {
+				if latestCompatible := platform.GetLatestCompatibleRelease(); latestCompatible != nil {
+					name := latestCompatible.Name
+					if latestCompatible.Deprecated {
+						name = fmt.Sprintf("[%s] %s", tr("DEPRECATED"), latestCompatible.Name)
+					}
+					t.AddRow(platform.Id, latestCompatible.Version, name)
```

Comment:
make an anonymous function to add a PlatformRelease in a table row (to be reused a few lines below)